src/AuthenticatedAPIClient/authInterface.js

```diff
@@ -0,0 +1,76 @@
+import Cookies from 'universal-cookie';
+import jwtDecode from 'jwt-decode';
+
+// Apply the auth-related properties and functions to the Axios API client.
+export default function applyAuthInterface(authenticatedAPIClient, authConfig) {
+  /* eslint-disable no-param-reassign */
+  authenticatedAPIClient.appBaseUrl = authConfig.appBaseUrl;
+  authenticatedAPIClient.authServiceUrl = authConfig.authServiceUrl;
+  authenticatedAPIClient.accessTokenCookieName = authConfig.accessTokenCookieName;
+  authenticatedAPIClient.csrfCookieName = authConfig.csrfCookieName;
+  authenticatedAPIClient.loginUrl = `${authConfig.authServiceUrl}/login`;
+  authenticatedAPIClient.logoutUrl = `${authConfig.authServiceUrl}/logout`;
+  authenticatedAPIClient.refreshAccessTokenEndpoint = `${authConfig.authServiceUrl}/user_api/v1/account/refresh_access_token/`;
+  /**
+   * We will not try to refresh an expired access token before
+   * making requests to these auth-related URLs.
+   */
+  authenticatedAPIClient.authUrls = [
+    authenticatedAPIClient.refreshAccessTokenEndpoint,
+  ];
+
+  authenticatedAPIClient.getAuthenticationState = () => {
+    const state = {};
+
+    const token = authenticatedAPIClient.getDecodedAccessToken();
+    if (token) {
+      state.authentication = {
```

Comment:
I think eventually it will contain security-related information i.e. authz info. I'm going to leave this as is for now. We can refactor later if things don't work out that way.